staging/src/k8s.io/component-helpers/scheduling/corev1/helpers.go

```diff
@@ -99,3 +99,167 @@ func getFilteredTaints(taints []v1.Taint, inclusionFilter taintsFilterFunc) []v1
 	}
 	return filteredTaints
 }
+
+// ContainerType represents the type of the container
+// A similar type is also defined in k8s.io/kubernetes/pkg/api/pod, but using it would introduce an unwanted
+// dependency
+type ContainerType = byte
+
+const (
+	// ContainerTypeContainers is for normal containers
+	ContainerTypeContainers ContainerType = 1 << iota
+	// ContainerTypeInitContainers is for init containers
+	ContainerTypeInitContainers
+)
+
+// PodResourcesOptions controls the behavior of PodRequests and PodLimits.
+type PodResourcesOptions struct {
+	// Reuse, if provided will be reused to accumulate resources and returned by the PodRequests or PodLimits functions.
+	Reuse v1.ResourceList
+	// InPlacePodVerticalScalingEnabled indicates that the in-place pod vertical scaling feature gate is enabled.
+	InPlacePodVerticalScalingEnabled bool
+	// ExcludeOverhead controls if pod overhead is excluded from the calculation.
+	ExcludeOverhead bool
+	// ContainerFn is called with the effective resources required for each container within the pod.
+	ContainerFn func(res v1.ResourceList, containerType ContainerType)
+}
+
+// GetContainerStatus extracts the status of container "name" from "statuses".
+// It returns true if "name" exists, else returns false.
+func GetContainerStatus(statuses []v1.ContainerStatus, name string) (v1.ContainerStatus, bool) {
```

Comment:
this is already in `pkg/api/v1/pod/util.go`. Are there some cycler reference issue?

Reply:
Yeah, but I just reworked it.  Since we're looking up every container, might as well pre-compute the lookup.